python/dask_cudf/dask_cudf/sorting.py

```diff
@@ -61,9 +61,9 @@ def _shuffle_group_2(df, cols, ignore_index, nparts):
 def _simple_shuffle(df, columns, npartitions, ignore_index=True):
 
     token = tokenize(df, columns)
-
+    simple_shuffle_group_token = "simple-shuffle-group-" + token
```

Comment:
We'll have to avoid same string concatenation in all the loops. Avoiding this saves 2/3rd of the time spent actually.

```python
startTime = datetime.now()
k = 1000
t = str(454768798089)
for i in range(1000):
    y = [ "helloworld" + t
        for i in range(k)
        for j in range(100)
    ]
print("old",(datetime.now() - startTime).total_seconds())


startTime = datetime.now()
k = 1000
x = "helloworld" + str(454768798089)
for i in range(1000):
    ll = [ x
        for i in range(k)
        for j in range(100)
    ]
print("new",(datetime.now() - startTime).total_seconds())
```

```python
old 10.286778
old 3.931635
```

Reply:
It would be good to see if there's similar opportunities to this available in upstream dask because that's a huge speedup and likely scales with the size of the task graph.

Reply:
Can also materialize one list for `range(k)` as well as another for `range(100)` to save on materializing those within the iteration as well.

Reply:
> It would be good to see if there's similar opportunities to this available in upstream dask because that's a huge speedup and likely scales with the size of the task graph.

Yep, checking in `dask` upstream if there are any such instances.